dev-client/App.tsx

```diff
@@ -5,34 +5,58 @@
  * @format
  */
 
-import React from 'react';
-import { StyleSheet, View } from 'react-native';
+import React, {useEffect} from 'react';
+import {PermissionsAndroid, StyleSheet, View} from 'react-native';
 
 import Mapbox from '@rnmapbox/maps';
-import { Button, NativeBaseProvider, Text } from 'native-base';
-import { theme } from './theme';
+import {Button, NativeBaseProvider, Text} from 'native-base';
+import {theme} from './theme';
 import LoginView from './components/Login';
-
-
+import SiteMap from './components/SiteMap';
+import {DisplaySite} from './datatypes/sites';
 
 Mapbox.setAccessToken(
   'pk.eyJ1Ijoic2hyb3V4bSIsImEiOiJjbGY4bW8wbGEwbDJnM3FsN3I1ZzBqd2kzIn0.2Alc4o911ooGEtnObLpOUQ',
 );
 
-
 function App(): JSX.Element {
+  useEffect(() => {
+    const locationGranted = PermissionsAndroid.request(
+      PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
+    )
+      .then(granted => {
+        if (granted == PermissionsAndroid.RESULTS.GRANTED) {
+          console.log('You can use the camera');
+        } else {
+          // TODO: What to do if rejected?
+          console.error('You cannot use the camera');
+        }
+      })
+      .catch(err => {
+        // TODO: What to do on error?
+        console.error(err);
+      });
+  });
```

Comment:
question: do we need to do this here as well as declaring the permissions in the manifest?

Reply:
Yes, for Android there are a few permissions that need to be requested explicitly: [AndroidPermissions](https://reactnative.dev/docs/permissionsandroid).

> PermissionsAndroid provides access to Android M's new permissions model. The so-called "normal" permissions are granted by default when the application is installed as long as they appear in AndroidManifest.xml. However, "dangerous" permissions require a dialog prompt. You should use this module for those permissions.